### DISCUSSION TITLE
Dataset for speech enhancement

### DISCUSSION THREAD

**Discussion by seastar105** [speechbrain/speechbrain#1889] [Q&A]
It seems there's two way to make variable lenght speech data batching, slice speech into fixed length, or padding.

speechbrain is using padded batch. but there could be some disadvantage of padding when speech enhancement training. which is better way to train speech enhancement model? slice? padding?

**Comment by pplantinga** [comment]
It is not clear to me which is better, and it may depend on your use case. If there will sometimes be silence in the audio, padding may actually improve the system by expanding the training domain to include silent segments.
